src/core/astish.js

```diff
@@ -1,4 +1,4 @@
-let newRule = /(?:([\u0080-\uFFFF\w-%@]+) *:? *([^{;]+?);|([^;}{]*?) *{)|(}\s*)/g;
+let newRule = /(?:([\u0080-\uFFFF\w-%@]+) *:? *((?:'(?:\\'|.)*?'|"(?:\\"|.)*?"|\([^\)]*?\)|[^{};])+)(?=[;}])|([^;}{]*?) *{)|(}\s*)/g;
```

Comment:
```suggestion
let newRule = /(?:([\u0080-\uFFFF\w-%@]+) *:? *((?:\([^\)]+?\)|[^{};])+)(?=[;}])|([^;}{]*?) *{)|(}\s*)/g;
```

Thanks for the ping! So deep diving into the rule was wondering why the need to target the `'|"` chars? 🤔 so did a quick test, remove them and tests are passing. So was there any particular caveat that you were chasing here?

Reply:
Oh, I'll explain the RegEx. Also you can inspect this regex via https://regex101.com/r/MAfztK/1 link. It's pretty good.

Reply:
With the above the changes are only `+16B` which is not bad at all.

Reply:
@ksenginew yeah saw the regex101 explanation. Here's the regex101 example with the trimmed down rule https://regex101.com/r/3ylRk5/1

Reply:
The initial regex rule matches the `'|"` and also the `(|)` braces. So that was my question if there's another case where this leads to issues if we remove the matching for `'|"`?

Reply:
@cristianbote your regex fails for `content` property.
```css
.close-me::after {
    content: "closing with ; now it's closed"; /* failed here */
}
```

**My solution will not fail in rare cases. But your solution looks enough.**

This is our OLD regular expression.
```py
/(?:([\u0080-\uFFFF\w-%@]+) *:? *([^{;]+?);|([^;}{]*?) *{)|(}\s*)/g
```
We can split it -`.split("|")`
- declaration
    ```py
   (?:([\u0080-\uFFFF\w-%@]+) *:? *([^{;]+?);
    ```
- selector or at rule
    ```py
    ([^;}{]*?) *{)
    ```
- last `}` character
    ```py
    (}\s*)
    ```

We want to change the part of declaration
```py
(?:([\u0080-\uFFFF\w-%@]+) *:? *([^{;]+?);
```
- property
    ```
    (?:([\u0080-\uFFFF\w-%@]+)
    ```
- `:` mark. It's optional to handle `@charset`
    ```css
     *:? *
    ```
- value
    ```py
   ([^{;]+?);
   ```

We should choose value 
```py
 ([^{;]+?);
```

I had changed it to
```diff
- ([^{;]                                             +?   );
+ ((?:'(?:\\'|.)*?'|"(?:\\"|.)*?"|\([^\)]*?\)|[^{};])+)(?=[;}])
```
inside this group
```py
((?:'(?:\\'|.)*?'|"(?:\\"|.)*?"|\([^\)]*?\)|[^{};])+)
```
There are 3 uncaptured groups added.
```py
'(?:\\'|.)*?'
```
```py
"(?:\\"|.)*?"
```
```py
\([^\)]*?\
```
```py
[^{};])  # already existed
```
These 3 uncp... groups handles double quoted strings, single quoted strings and CSS function calls.
```py
'(?:\\'|.)*?'
```

Can we take a last decision. then I can optimize regex and commit it.

Reply:
Oh, that makes sense now, those are for the `content` property which can have a variation of things inside it and not mention defined differently based on the input format.

Ok let's merge this and we can look for things to minify later on. Great work @ksenginew!